Library/Homebrew/extend/blank.rb

```diff
@@ -126,30 +153,43 @@ def blank?
       begin
         BLANK_RE.match?(self)
       rescue Encoding::CompatibilityError
-        ENCODED_BLANKS[self.encoding].match?(self)
+        Regexp.new(BLANK_RE.source.encode(encoding), BLANK_RE.options | Regexp::FIXEDENCODING).match?(self)
```

Comment:
Do we know how much of an edge case this is? I think we should cache the encodings just in case if we're not sure.

Reply:
I don't think I've ever encountered this shape of error, but I'm happy to employ a non-concurrent cache for this.